Add 'google_folder_iam_binding' and 'google_folder_iam_member' resources

ewbankkit opened the pull request:
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/1075.

Acceptance tests:
* `google_folder_iam_binding` resource
```
make testacc TEST=./google/ TESTARGS='-run=TestAccGoogleFolderIamBinding_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccGoogleFolderIamBinding_ -timeout 120m
=== RUN   TestAccGoogleFolderIamBinding_basic
=== RUN   TestAccGoogleFolderIamBinding_multiple
=== RUN   TestAccGoogleFolderIamBinding_multipleAtOnce
=== RUN   TestAccGoogleFolderIamBinding_update
=== RUN   TestAccGoogleFolderIamBinding_remove
--- PASS: TestAccGoogleFolderIamBinding_basic (17.41s)
--- PASS: TestAccGoogleFolderIamBinding_remove (19.18s)
--- PASS: TestAccGoogleFolderIamBinding_update (24.55s)
--- PASS: TestAccGoogleFolderIamBinding_multipleAtOnce (18.64s)
--- PASS: TestAccGoogleFolderIamBinding_multiple (23.73s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	103.516s
```
* `google_folder_iam_member` resource
```
make testacc TEST=./google/ TESTARGS='-run=TestAccGoogleFolderIamMember_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccGoogleFolderIamMember_ -timeout 120m
=== RUN   TestAccGoogleFolderIamMember_basic
=== RUN   TestAccGoogleFolderIamMember_multiple
=== RUN   TestAccGoogleFolderIamMember_remove
--- PASS: TestAccGoogleFolderIamMember_basic (17.47s)
--- PASS: TestAccGoogleFolderIamMember_remove (19.73s)
--- PASS: TestAccGoogleFolderIamMember_multiple (22.66s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	59.874s
```

ewbankkit commented:
Regression tests for code refactored as part of this PR:
* `google_active_folder` data source
```
make testacc TEST=./google/ TESTARGS='-run=TestAccDataSourceGoogleActiveFolder'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccDataSourceGoogleActiveFolder -timeout 120m
=== RUN   TestAccDataSourceGoogleActiveFolder
--- PASS: TestAccDataSourceGoogleActiveFolder (13.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	13.197s
```
* `google_folder_iam_policy` resource
```
make testacc TEST=./google/ TESTARGS='-run=TestAccGoogleFolderIamPolicy_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccGoogleFolderIamPolicy_ -timeout 120m
=== RUN   TestAccGoogleFolderIamPolicy_basic
=== RUN   TestAccGoogleFolderIamPolicy_update
--- PASS: TestAccGoogleFolderIamPolicy_basic (15.52s)
--- PASS: TestAccGoogleFolderIamPolicy_update (17.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	33.507s
```

ndmckinley commented:
Some test failures, which I have to imagine are flakes - rerunning to deflake now.

ndmckinley commented:
Hm, they're persistent:
```
------- Stdout: -------
=== RUN   TestAccGoogleFolderIamBinding_update
--- FAIL: TestAccGoogleFolderIamBinding_update (24.55s)
    testing.go:513: Step 3 error: Check failed: Check 1/1 error: IAM policy for folder "terraform-se307rl6m0" had no role "roles/compute.instanceAdmin"
FAIL
```

Any thoughts?

ewbankkit commented:
I rebased to remove the merge conflict and re-ran the tests:

```
make testacc TEST=./google/ TESTARGS='-run=TestAccGoogleFolderIamBinding_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccGoogleFolderIamBinding_ -timeout 120m
=== RUN   TestAccGoogleFolderIamBinding_basic
=== RUN   TestAccGoogleFolderIamBinding_multiple
=== RUN   TestAccGoogleFolderIamBinding_multipleAtOnce
=== RUN   TestAccGoogleFolderIamBinding_update
=== RUN   TestAccGoogleFolderIamBinding_remove
--- PASS: TestAccGoogleFolderIamBinding_basic (17.71s)
--- PASS: TestAccGoogleFolderIamBinding_multipleAtOnce (18.83s)
--- PASS: TestAccGoogleFolderIamBinding_remove (19.21s)
--- PASS: TestAccGoogleFolderIamBinding_multiple (20.58s)
--- PASS: TestAccGoogleFolderIamBinding_update (23.54s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	29.782s
```

```
make testacc TEST=./google/ TESTARGS='-run=TestAccGoogleFolderIamMember_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google/ -v -run=TestAccGoogleFolderIamMember_ -timeout 120m
=== RUN   TestAccGoogleFolderIamMember_basic
=== RUN   TestAccGoogleFolderIamMember_multiple
=== RUN   TestAccGoogleFolderIamMember_remove
--- PASS: TestAccGoogleFolderIamMember_basic (16.62s)
--- PASS: TestAccGoogleFolderIamMember_remove (19.09s)
--- PASS: TestAccGoogleFolderIamMember_multiple (20.08s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	20.368s
```

I added additional information to the error message for the failing case.

ndmckinley commented:
I'll try it again.

ndmckinley commented:
Tests passed after that commit, merging.
